src/dmd/dtoh.d

```diff
@@ -0,0 +1,1256 @@
+                {
+                    visit(cast(AST.Dsymbol)ad);
+                    return;
+                }
+                buf.writestring("typedef ");
+                typeToBuffer(t, ad.ident);
+                buf.writestring(";\n");
+                if (!adparent)
+                    buf.printf("\n");
+                return;
+            }
+            if (!ad.aliassym)
+            {
+                ad.print();
+                assert(0);
+            }
+            if (auto ti = ad.aliassym.isTemplateInstance())
+            {
+                visitTi(ti);
+                return;
+            }
+            if (auto sd = ad.aliassym.isStructDeclaration())
+            {
+                buf.writestring("typedef ");
+                sd.type.accept(this);
+                buf.writestring(" ");
+                buf.writestring(ad.ident.toChars());
+                buf.writestring(";\n");
+                if (!adparent)
+                    buf.printf("\n");
+                return;
+            }
+            indent();
+            buf.printf("// ignored %s %s\n", ad.aliassym.kind(), ad.aliassym.toPrettyChars());
+        }
+
+        override void visit(AST.AnonDeclaration ad)
+        {
+            buf.writestring(ad.isunion ? "union" : "struct");
+            buf.writestring("\n{\n");
+            foreach (s; *ad.decl)
+            {
+                s.accept(this);
+            }
+            buf.writestring("};\n");
+        }
+
+        private bool memberField(AST.VarDeclaration vd)
+        {
+            if (!vd.type || !vd.type.deco || !vd.ident)
+                return false;
+            if (!vd.isField())
+                return false;
+            if (vd.type.ty == AST.Tfunction)
+                return false;
+            if (vd.type.ty == AST.Tsarray)
+                return false;
+            return true;
+        }
+
+        override void visit(AST.StructDeclaration sd)
+        {
+            if (sd.isInstantiated())
+                return;
+            if (cast(void*)sd in visited)
+                return;
+            if (!sd.type || !sd.type.deco)
+                return;
+            if (sd.getModule() && !sd.getModule().isFrontendModule())
+                return;
+            visited[cast(void*)sd] = true;
+
+            if (sd.alignment == 1)
+                buf.writestring("#pragma pack(push, 1)\n");
+            buf.writestring(sd.isUnionDeclaration() ? "union " : "struct ");
+            buf.writestring(sd.ident.toChars());
+            if (sd.members)
+            {
+                buf.writestring("\n{\n");
+                auto save = adparent;
+                adparent = sd;
+                foreach (m; *sd.members)
+                {
+                    m.accept(this);
+                }
+                adparent = save;
+                // Generate default ctor
+                buf.printf("    %s(", sd.ident.toChars());
+                buf.printf(") {");
```

Comment:
AFAICT the header files currently don't have this. Why was generating a default ctor necessary?

Reply:
You're thinking classes.  This is for structs, and we do have this in some places, such as `Loc`.